runtime/compiler/ilgen/Walker.cpp

```diff
@@ -6973,40 +6961,33 @@ TR_J9ByteCodeIlGenerator::storeInstance(int32_t cpIndex)
 
    TR_ResolvedJ9Method * owningMethod = static_cast<TR_ResolvedJ9Method*>(_methodSymbol->getResolvedMethod());
 
-   if (TR::Compiler->om.areFlattenableValueTypesEnabled())
+   if (owningMethod->isFieldNullRestricted(comp(), cpIndex, false /* isStatic */, true /* isStore */))
       {
-      if (!TR::Compiler->om.isQDescriptorForValueTypesSupported())
+      if (!isFieldResolved(comp(), owningMethod, cpIndex, true))
+         {
+         abortForUnresolvedValueTypeOp("putfield", "field");
+         }
+      else if (owningMethod->isFieldFlattened(comp(), cpIndex, _methodSymbol->isStatic()))
```

Comment:
As with the call to `isFieldNullRestricted`, I think this should be passing `false` as the third argument to `isFieldFlattened` rather than the result of `_methodSymbol->isStatic()`.